### ISSUE TITLE
BUG: writeCarrierDataToDatabase.py writes -inf for acceleration sometimes

### ISSUE THREAD

**Issue by danielreischl** [AMOSus/amos-ss16-proj6#31]
Hey,

we have a small issue with the acceleration calculation. If the the distance between two timestamps is 0, the acceleration is -inf. This is based on the formula we are using:

(SpeedEnd \* SpeedEnd - SpeedBegin \* SpeedBegin))/distance \* 2  (Dividing by 0 isn't good :D)

Should we replace these values with 0? Or should compressInitialData.py compress rows when the carrier isn't moving?

**Comment by ghost** [comment]
Lets wait until compressInitialData.py is finalized as in issue #30

**Comment by TobiDo** [comment]
Daniel, could you clarify this, I don't get it: we shouldn't get the same timestamp twice. This will either create an inconsistency or will simply be redundant. It has (in principle) nothing to do with the carrier moving or not.

**Comment by ghost** [comment]
compressInitialData.py is finalized. There should never be the case where we have 2 same Time stamps in a row because of 2 reasons:
- The script checks before processing a timestamp if it is the same as before:
  as in line 73 - 76

```
# If the timestamp is the same as previously, the data is not recorded
    # This only happens when pushing carries which should not happen in the first place
    if time == carrierData[carrier - 1][0][currentPositionAtCarrierData[carrier - 1] - 1]:
        return
```
- Compress data aggregates the data over multiple stamps, therefore it should also not be the case.

**Comment by danielreischl** [comment]
@TobiDo Not the timestamp but the the position is the same. Distance is Calculted by Position @ Timestamp2 - Position @ Timestamp1

Which means if the the carrier isn't moving between two timestamps, the formula tries to divide by 0:

(SpeedEnd \* SpeedEnd - SpeedBegin \* SpeedBegin) / distance \* 2

**Comment by ghost** [comment]
The position will definitely occur if the carrier didn't move during that time. My suggestion:

`(SpeedEnd * SpeedEnd - SpeedBegin * SpeedBegin) / distance * 2 if distance != 0 else 0`

I don't quite understand this formula however 🐒

**Comment by danielreischl** [comment]
![image](https://cloud.githubusercontent.com/assets/18442099/15473244/5ffb520e-20ff-11e6-8e68-495168ec5d6e.png)

That's the formula I used.

Thx to: http://formulas.tutorvista.com/physics/acceleration-formula.html

**Comment by ghost** [comment]
Ok but then just work in my suggested fix?

**Comment by TobiDo** [comment]
Sorry I didn't read it carefully. Why do we not just calculated acceleration as (numerical) first derivative of speed:
(Speed2 - Speed1)/ (time2 - time1).
Then the problem goes away. What we do now is 
(Speed2 \* Speed2 - Speed1\* Speed1)/ (2 \* distance) = (Speed2-Speed1) \* (Speed2 + Speed1) / (2  *distance)
(Speed1 + Speed2)/2 is the average of the two speed values, so (Speed2 + Speed1) / (2 \* distance) is roughly the time difference. So as one should have hoped, there's not a big difference. Rene's suggestion is ok, but I think (Speed2 - Speed1)/ (time2 - time1) is cleaner and easier to understand.

**Comment by danielreischl** [comment]
yes makes sense, I will change the formula to (Speed2 - Speed1)/ (time2 - time1)

**Comment by danielreischl** [comment]
Fixed with: 4e7ca1ed8b380f06047369b23a0026ab2f4ad4a1
